### PR TITLE
netopeer2 with new libnetconf2 API

### PR DESCRIPTION
**Pull request by Roytak**
Initial changes for netopeer2 in order to use new libnetconf2 API.

Some of the most notable changes include:
- removal of configuration callbacks, it's handled by ln2
- removal of some redundant command line opts
- removal of some np2's YANG modules, they are loaded via ln2's schema dir
- the SSH and TLS options have been merged into one, that is SSH_TLS

### CODE REVIEW

File: tests/np_test.c [hunk 1]
```diff
@@ -221,9 +232,7 @@ np_glob_setup_np2(void **state, const char *test_name, const char **modules)
         close(fd);
 
         /* exec server listening on a unix socket */
```
Comment (michalvasko) [inline tests/np_test.c:234]: There is a UNIX socket but not set up here so adjust the comment.

**Review by michalvasko — CHANGES_REQUESTED** [review]

File: scripts/common.sh [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+    if [ -z "$SYSREPOCFG" ]; then
+    	echo "$0: Unable to find sysrepocfg executable." >&2
+        exit 1
+    fi
+}
+
+# get path to the sysrepoctl executable
+function SYSREPOCTL_GET_PATH() {
+	if [ -n "$SYSREPOCTL_EXECUTABLE" ]; then
```
Comment (michalvasko) [inline scripts/common.sh:60]: Indent.

File: scripts/merge_config.sh [hunk 3]
```diff
@@ -2,20 +2,69 @@
...
+            IDX=$((IDX + 1))
+        fi
+    done < "$AUTHORIZED_KEYS_FILE"
+
+    # append the ending tags
+    AUTH_CONFIG="${AUTH_CONFIG}
+                            </inline-definition>
+                        </public-keys>"
+
+    echo "Creating user \"${CURRENT_USER}\" that can authenticate with a key pair from his authorized_keys..."
```
Comment (michalvasko) [inline scripts/merge_config.sh:60]: The message (and the one below) is a bit inaccurate, the user is not being created, only added into the server configuration.

File: scripts/remove.sh [hunk 4]
```diff
@@ -1,54 +1,31 @@
 #!/usr/bin/env bash
 # {% raw %} - jinja2 command to not process "{#" in the script
```
Comment (michalvasko) [inline scripts/remove.sh:2]: This can be removed, obsolete now. At the script end, too.

File: src/netconf_server.c [hunk 5]
```diff
@@ -32,517 +35,153 @@
...
 
-    return SR_ERR_OK;
+    return 0;
 }
 
-/* /ietf-netconf-server:netconf-server/listen/endpoint/ * /tcp-server-parameters */
 int
-np2srv_endpt_tcp_params_cb(sr_session_ctx_t *session, uint32_t UNUSED(sub_id), const char *UNUSED(module_name),
-        const char *xpath, sr_event_t UNUSED(event), uint32_t UNUSED(request_id), void *UNUSED(private_data))
+np2srv_pubkey_auth_cb(const struct nc_session *session, ssh_key key, void *UNUSED(user_data))
```
Comment (michalvasko) [inline src/netconf_server.c:84]: Server (libnetconf2) should use the authorized keys configured for the client, not read them from the system or am I missing something?

File: tests/np_test.c [hunk 6]
```diff
@@ -167,6 +175,10 @@ np_glob_setup_np2(void **state, const char *test_name, const char **modules)
         SETUP_FAIL_LOG;
         return 1;
     }
+    if (unsetenv("LN2_MODULE_DIR")) {
```
Comment (michalvasko) [inline tests/np_test.c:178]: Missing unset for `NP2_SCRIPTS_DIR`.

**Review by michalvasko — CHANGES_REQUESTED** [review]

**Comment by Roytak** [comment]
Figured there is no need for the files `netconf_server.c` and `netconf_server.h` if there's only one function definition (the libnetconf2 callback) in them, so removed them.

**Review by michalvasko — APPROVED** [review]

**Pull request merged**